syllabus/4-Smart_Contracts/6-Wasm_ink!_slides.md

```diff
@@ -1054,235 +1149,1255 @@ Notes:
+* **Generalized Frontrunners** are bots looking for *any* profitable transactions submitted to the mempool.
+* *Ethereum is a Dark Forest, Dan Robinson and Georgios Konstantopoulos* **[August 2020]**
+
+Notes:
+* Cixin Liu [Si-Szin Lju]
+* who's familiar with these novels?
+* The Dark Forest is a 2-nd book in the sci-fi series Remembrance of the Earth's Past
+* Publicly identifying someone else’s location is as good as directly destroying them
+* These Frontrunners work by copying it and replacing the address with their own
+  * They can even execute the transaction in a sandbox and submit just the profitable internal transactions as their own.
+  * One encounter with such a bot was described by these two researchers.
+
+---
+
+## Sidenote: The Dark Forest
+
+<img style="margin-top: 10px;margin-bottom: 10px" width="900" src="./img/ink/kurzgesagt.jpg" />
+
+credit: **Kurzgesagt** : *Why We Should NOT Look For Aliens - The Dark Forest*
+
+* The Dark Forest theory is (one of) a solution to the Fermi's paradox
+* Great explanation in this episode of the Kurzgesagt
+
+Notes:
+* Sidenote for the curious
+
+---
+
+## MEV: The Dark Forest
+
+* Someone had asked on the *#support* channel of Uniswap whether it was possible to recover liquidity tokens that were erronously sent to the <font color="#8d3aed">[liquidity token base contract](https://github.com/Uniswap/v2-core/blob/master/contracts/UniswapV2Pair.sol#L140)</font>.
+* This meant *anyone* who calls *burn* on the Uniswap contract, passing own address, would receive the extra tokens (worth *USD 12K*).
+* **Dan Robinson**, a researcher and a white-hat hacker offered to help.
+
+Notes:
+* Not even a bug, jusy a by-product of how UniswapV2 is designed
+* For details I refer you to the original article
+* Only that he knew this wouldn't be as easy as calling burn and returning the tokens to the owner
+* ... because of the monsters in the forest
+
+---
+
+## MEV: The Dark Forest
+
+<img style="margin-top: 10px;margin-bottom: 10px" width="900" src="./img/ink/flashboys.png" />
+
+* In another article, *Phil Daian et al* talked about how one particular species frontrunning bots, called **Generalized Frontrunners** scans the mempool for profitabe transactions.
+* If someone just submitted a *burn* transaction to the mempool the Dark Forest bots would be imediately alerted.
+
+Notes:
+* Dan was well aware this would not be easy
+* it was a ticking time bomb
+* They had to desing an obfuscated rescue method to try and not alert the bots about the free money.
+
+---
+
+## The Dark Forest: To The Rescue
+
+* Dan Robinson recruited the help of fellow engineers (i.e. [Georgios Konstantopoulos]() and [Alberto Cuesta Canada]() and a few others) and they came up with the following solution:
+
+<div style="font-size: 0.62em;">
+
+```Solidity [1-7 | 9-33 | 35-47]
```

Comment:
Your code extends outside of the screen. Maybe reduce the size of the text above, or place it in different slide

Reply:
it's fine - i scrolls on animation to the relevant parts